sdk/dns/Azure.ResourceManager.Dns/src/Customization/Models/Zones.cs

```diff
@@ -0,0 +1,71 @@
+        /// <param name="etag"> The etag of the zone. </param>
+        /// <param name="maxNumberOfRecordSets"> The maximum number of record sets that can be created in this DNS zone.  This is a read-only property and any attempt to set this value will be ignored. </param>
+        /// <param name="numberOfRecordSets"> The current number of record sets in this DNS zone.  This is a read-only property and any attempt to set this value will be ignored. </param>
+        /// <param name="nameServers"> The name servers for this DNS zone. This is a read-only property and any attempt to set this value will be ignored. </param>
+        /// <param name="zoneType"> The type of this DNS zone (Public or Private). </param>
+        /// <param name="registrationVirtualNetworks"> A list of references to virtual networks that register hostnames in this DNS zone. This is a only when ZoneType is Private. </param>
+        /// <param name="resolutionVirtualNetworks"> A list of references to virtual networks that resolve records in this DNS zone. This is a only when ZoneType is Private. </param>
+        internal Zone(string id, string name, string type, string location, IDictionary<string, string> tags, string etag, long? maxNumberOfRecordSets, long? numberOfRecordSets, IReadOnlyList<string> nameServers, ZoneType? zoneType, IList<SubResource> registrationVirtualNetworks, IList<SubResource> resolutionVirtualNetworks) : base(id, name, type, location, tags)
+        {
+            Etag = etag;
+            MaxNumberOfRecordSets = maxNumberOfRecordSets;
+            NumberOfRecordSets = numberOfRecordSets;
+            NameServers = nameServers;
+            ZoneType = zoneType;
+            RegistrationVirtualNetworks = registrationVirtualNetworks;
+            ResolutionVirtualNetworks = resolutionVirtualNetworks;
+        }
+
+        /// <summary> The etag of the zone. </summary>
+        public string Etag { get; set; }
+        /// <summary> The maximum number of record sets that can be created in this DNS zone.  This is a read-only property and any attempt to set this value will be ignored. </summary>
+        public long? MaxNumberOfRecordSets { get; }
+        /// <summary> The current number of record sets in this DNS zone.  This is a read-only property and any attempt to set this value will be ignored. </summary>
+        public long? NumberOfRecordSets { get; }
+        /// <summary> The name servers for this DNS zone. This is a read-only property and any attempt to set this value will be ignored. </summary>
+        public IReadOnlyList<string> NameServers { get; }
+        /// <summary> The type of this DNS zone (Public or Private). </summary>
+        public ZoneType? ZoneType { get; }
+        /// <summary> A list of references to virtual networks that register hostnames in this DNS zone. This is a only when ZoneType is Private. </summary>
+        public IList<SubResource> RegistrationVirtualNetworks { get; }
```

Comment:
Do we need [Obsolete] on these two properties?

Reply:
whoops yes, let me add that

Reply:
they are only read-only, so they would only come back from a GET on a private DNS zone

Reply:
compile warning not error.

Reply:
I think [Obsolete] can be applied here. If user are using these properties, they will get a compiler warning which is what we wanted.

Reply:
Well depends on how one builds I suppose. Dotnet core on Ubuntu throws a compile error. 

I will make this change.